sched/adaptdl_sched/allocator.py

```diff
@@ -102,62 +168,71 @@ async def _find_nodes(self):
         node_template = NodeInfo(max_resources, True)
         return node_infos, node_template
 
+    def _get_job_info(self, job):
+        job["spec"]["template"]["spec"] = \
+            set_default_resources(job["spec"]["template"]["spec"])
+        resources = get_pod_requests(job["spec"]["template"]["spec"])
+        hints = job.get("status", {}).get("train", {})
+        max_replicas = max(2 * hints.get("maxProfiledReplicas", 0), 1)
+        if job["spec"].get("maxReplicas"):
+            max_replicas = min(max_replicas, job["spec"]["maxReplicas"])
+        min_replicas = job["spec"].get("minReplicas", 0)
+        # max_replicas should be greater or equal to min_replicas
+        max_replicas = max(max_replicas, min_replicas)
+        preemptible = job["spec"].get("preemptible", True)
+        if {"perfParams", "initBatchSize"} <= hints.keys() and preemptible:
+            max_batch_size = (hints.get("maxBatchSize")
+                              or hints["initBatchSize"])
+            if hints.get("localBszBounds"):
+                min_local_bsz = hints["localBszBounds"][0] or 1
+                # Make sure max_batch_size / replicas >= min_local_bsz
+                if max_batch_size < min_local_bsz * max_replicas:
+                    max_replicas = int(max_batch_size / min_local_bsz)
+            perf_params = PerfParams(*[hints["perfParams"][k]
+                                       for k in PERF_PARAMS.keys()])
+            if "gradParams" in hints:
+                grad_params = GradParams(hints["gradParams"]["norm"],
+                                         hints["gradParams"]["var"])
+            else:
+                grad_params = GradParams(0.0, 1.0)
+            goodput_fn = GoodputFunction(perf_params, grad_params,
+                                         hints["initBatchSize"])
+            speedup_fn = SpeedupFunction(
+                goodput_fn,
+                hints.get("maxBatchSize"),
+                hints.get("localBszBounds"),
+                hints.get("gradientAccumulation", False))
+        else:
+            speedup_fn = lambda n, r: r  # noqa: E731
+        creation_ts = dateutil.parser.isoparse(
+                job["metadata"]["creationTimestamp"])
+        return JobInfo(
+                resources, speedup_fn, creation_ts, min_replicas,
+                max_replicas, preemptible)
+
     async def _find_jobs_and_allocations(self):
         job_list = await self._objs_api.list_namespaced_custom_object(
             "adaptdl.petuum.com", "v1", "", "adaptdljobs")
         job_infos = {}
         allocations = {}
+
         for job in job_list["items"]:
             if job.get("status", {}).get("phase") \
                     not in ["Pending", "Running", "Starting", "Stopping"]:
+                LOG.info("unkown status:", job,
+                         job.get("status", {}).get("phase"))
```

Comment:
Don't think we want to log anything here since this can print out a lot of stuff, e.g. for all completed jobs